plugins/woocommerce/includes/class-wc-cart-session.php

```diff
@@ -95,33 +94,33 @@ public function init() {
 	 * @since 3.2.0
 	 */
 	public function get_cart_from_session() {
+		/**
+		 * Fires when cart is loaded from session.
+		 *
+		 * @since 3.2.0
+		 */
 		do_action( 'woocommerce_load_cart_from_session' );
-		$this->cart->set_totals( WC()->session->get( 'cart_totals', null ) );
+
+		$cart        = (array) array_filter( WC()->session->get( 'cart', array() ) );
+		$cart_totals = WC()->session->get( 'cart_totals', null );
+
+		$this->cart->set_totals( $cart_totals );
 		$this->cart->set_applied_coupons( WC()->session->get( 'applied_coupons', array() ) );
 		$this->cart->set_coupon_discount_totals( WC()->session->get( 'coupon_discount_totals', array() ) );
 		$this->cart->set_coupon_discount_tax_totals( WC()->session->get( 'coupon_discount_tax_totals', array() ) );
 		$this->cart->set_removed_cart_contents( WC()->session->get( 'removed_cart_contents', array() ) );
 
-		$update_cart_session = false; // Flag to indicate the stored cart should be updated.
-		$order_again         = false; // Flag to indicate whether this is a re-order.
-		$cart                = WC()->session->get( 'cart', null );
-		$merge_saved_cart    = (bool) get_user_meta( get_current_user_id(), '_woocommerce_load_saved_cart_after_login', true );
+		// Flag to indicate the stored cart should be updated. If cart totals are null, this will be true to calculate totals.
+		$update_cart_session = is_null( $cart_totals );
```

Comment:
I'm struggling to understand why this change was needed? as opposed to the previous one?

Reply:
Naming change?

Reply:
We had `$update_cart_session = false;` before, and below we would check that and null totals. Now it's just null totals, why? because in this code path, it will always be false?